examples/tv-app/tv-common/clusters/messages/MessagesManager.cpp

```diff
@@ -18,39 +18,152 @@
+void MessagesManager::HandlePresentMessagesRequest(const ByteSpan & messageId, const MessagePriorityEnum & priority,
+                                                   const BitMask<MessageControlBitmap> & messageControl,
+                                                   const DataModel::Nullable<uint32_t> & startTime,
+                                                   const DataModel::Nullable<uint16_t> & duration, const CharSpan & messageText,
+                                                   const Optional<DataModel::DecodableList<MessageResponseOption>> & responses)
 {
-    Message message{
-        // TODO: Enable id
-        chip::ByteSpan(), priority, messageControl, startTime, duration,
-        // TODO: Enable text
-        chip::CharSpan()
-        // TODO: Convert responses to Optional<chip::app::DataModel::List<const
-        // chip::app::Clusters::Messages::Structs::MessageResponseOptionStruct::Type>> message.responses = responses;
-    };
-
-    mMessages.push_back(message);
+    std::vector<char *> needToFree;
+    std::vector<MessageResponseOption *> optionToFree;
+
+    uint8_t * messageIdBuffer = nullptr;
+    char * messageTextBuffer  = new char[messageText.size()];
+    VerifyOrExit(messageTextBuffer != nullptr,
+                 ChipLogProgress(Controller, "HandlePresentMessagesRequest messageTextBuffer alloc failed"));
+    needToFree.push_back(messageTextBuffer);
+    memcpy(messageTextBuffer, messageText.data(), messageText.size());
+
+    ChipLogProgress(Controller, "HandlePresentMessagesRequest message:%s size:%lu", messageTextBuffer, messageText.size());
+
+    messageIdBuffer = new uint8_t[messageId.size()];
+    VerifyOrExit(messageIdBuffer != nullptr,
+                 ChipLogProgress(Controller, "HandlePresentMessagesRequest messageIdBuffer alloc failed"));
+    memcpy(messageIdBuffer, messageId.data(), messageId.size());
+
+    if (responses.HasValue())
+    {
+        size_t size    = 0;
+        CHIP_ERROR err = responses.Value().ComputeSize(&size);
+        VerifyOrExit(err == CHIP_NO_ERROR, ChipLogProgress(Controller, "HandlePresentMessagesRequest size check failed"));
+
+        MessageResponseOption * optionArray = new MessageResponseOption[size];
+        VerifyOrExit(optionArray != nullptr,
+                     ChipLogProgress(Controller, "HandlePresentMessagesRequest MessageResponseOption alloc failed"));
+        optionToFree.push_back(optionArray);
+
+        int counter = 0;
+        auto iter   = responses.Value().begin();
+        while (iter.Next())
+        {
+            auto & response = iter.GetValue();
+
+            ChipLogProgress(Controller, "HandlePresentMessagesRequest option id:%u", response.messageResponseID.ValueOr(0));
+
+            if (response.messageResponseID.HasValue())
+            {
+                optionArray[counter].messageResponseID = Optional<uint32_t>(response.messageResponseID.ValueOr(0));
```

Comment:
This doesn't make sense.  Shouldn't we be checking that the field is there if we support RESP and missing if we are not?  Setting multiple of them to 0 violates the spec: all the values must be unique.

Reply:
added early exit when either field is missing